### PR TITLE
Creating and enabling system tests to verify creation of span links for distributed tracing

### PR DESCRIPTION
**Pull request by mhlidd**
## Motivation

With distributed tracing, there are scenarios where we do not have a connection between a span created in a different service and spans created in Datadog. Thus, when a span is propagated from a different service to Datadog and has a different `trace_id` from the local trace, we want to use a `span_link` in order to create a relationship between the spans.

[RFC](https://docs.google.com/document/d/1xacBonCyuVk95D-L1STXGqdLtsOuf0_3jG-138uXHnQ/edit?tab=t.0#bookmark=id.uwnmpxy1znny)

## Changes

Creating a new system test to check the existence of span links when spans are created with a w3c format precedence when spans are propagated through headers. Additionally, enabling the test that checks the existence of span links when spans are created with Datadog format precedence. 

[APMAPI-758](https://datadoghq.atlassian.net/browse/APMAPI-758)
## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

[APMAPI-758]: https://datadoghq.atlassian.net/browse/APMAPI-758?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by robertomonteromiguel — APPROVED** [review]

File: tests/parametric/test_headers_precedence.py [hunk 1]
```diff
@@ -786,6 +795,60 @@ def test_headers_precedence_propagationstyle_resolves_conflicting_contexts(self,
...
+                    ["x-datadog-tags", "_dd.p.tid=2222222222222222"],
+                    ["x-b3-traceid", "11111111111111110000000000000003"],
+                    ["x-b3-spanid", "a2fb4a1d1a96d312"],
+                    ["x-b3-sampled", "1"],
+                ],
+            ) as s1:
+                pass
+
+        traces = test_agent.wait_for_num_traces(1)
+        assert len(traces) == 1
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:830]:
```suggestion
```

nit: This is unnecessary,`wait_for_num_traces` will raise an Exception if 1 trace is not captured.

File: tests/parametric/test_headers_precedence.py [hunk 2]
```diff
@@ -786,6 +795,60 @@ def test_headers_precedence_propagationstyle_resolves_conflicting_contexts(self,
...
+        span = find_span_in_traces(traces, s1.trace_id, s1.span_id)
+
+        assert span["name"] == "trace_ids_do_not_match"
+        span_links = retrieve_span_links(span)
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:834]: I know the implementation details for python, ruby, and go so I can confidently say span links generated in these libraries are serialized in the order they are created. I am not sure if that's the case for all languages. This is a behavior that wasn't called out in the RFC and from a UI perspective it honestly doesn't matter. 

This is something to look out for as you enable this test for other languages. I don't think you need to make any changes now.

File: tests/parametric/test_headers_precedence.py [hunk 3]
```diff
@@ -786,6 +795,60 @@ def test_headers_precedence_propagationstyle_resolves_conflicting_contexts(self,
...
+        assert link1["attributes"] == {"reason": "terminated_context", "context_headers": "datadog"}
+        assert link1["flags"] == 1 | TRACECONTEXT_FLAGS_SET
+        assert link1["trace_id_high"] == 2459565876494606882
+        assert link1.get("tracestate") == None
+
+        link2 = span_links[1]
+        assert link2["trace_id"] == 3
+        assert link2["span_id"] == 11744061942159299346
+        assert link2["attributes"] == {"reason": "terminated_context", "context_headers": "b3multi"}
+        assert link2["flags"] == 1 | TRACECONTEXT_FLAGS_SET
```
Comment (mabdinur) [inline tests/parametric/test_headers_precedence.py:848]: nit: It would be nice if one of the headers set a sampled tag of `0 | TRACECONTEXT_FLAGS_SET`

**Review by mabdinur — APPROVED** [review]
LGTM

**Pull request merged**